### PR TITLE
Add QA-SRL-Bank 2.0 Reader 

### PR DESCRIPTION
**Pull request by ziqian98**
This PR fixes [ #231]. 

### Description of changes
A reader for QA-SRL-Bank 2.0 is added.

### Possible influences of this PR.
A new Reader for a new Dataset

### Test Conducted
Three lines of QA-SRL-Bank 2.0 dataset is load and tested.
Unit test Conducted

### CODE REVIEW

File: forte/data/readers/qasrl_reader.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    "QASRLReader"
+]
+
+
+class QASRLReader(PackReader):
+    r""":class:`QASRLReader` is designed to read in the QA-SRL Bank 2.0 dataset used
+        by Question-Answer driven Semantic Role Labeling (QA-SRL) annotations task.
+        The related paper can be found here
+        https://arxiv.org/pdf/1805.05377.pdf
+        The Original data format is somewhat complex, you can find it here:
```
Comment (hunterhector) [inline forte/data/readers/qasrl_reader.py:43]: You don't have to say "this is complex"

**Review by hunterhector — CHANGES_REQUESTED** [review]
Your reader only reads the sentences, where are the questions and labels?

**Comment by hunterhector** [comment]
If this is still a draft, you can convert it into a draft first.

File: forte/data/readers/qasrl_reader.py (hunk 1 above)
Comment (ziqian98) [reply forte/data/readers/qasrl_reader.py]: Fixed

**Comment by ziqian98** [comment]
Each line in the original dataset is like the structure:  https://github.com/uwnlp/qasrl-bank/blob/master/FORMAT.md
I read each line as one sentence, which contains one Question and Semantic Labels.

**Comment by hunterhector** [comment]
> Each line in the original dataset is like the structure: https://github.com/uwnlp/qasrl-bank/blob/master/FORMAT.md
> I read each line as one sentence, which contains one Question and Semantic Labels.

No, you have to mark Questions and Labels explicitly. Otherwise, when someone is using the reader, how can he know which one is the Question and which one is the Label? What you are doing here is reading plain text only.

**Review by hunterhector — CHANGES_REQUESTED** [review]
Please explicitly label the Question and Semantic Labels. Ask your teammates if you don't understand what this means. There is so much information in this data format and this reader only gives us the sentence.

**Comment by ziqian98** [comment]
> Please explicitly label the Question and Semantic Labels. Ask your teammates if you don't understand what this means. There is so much information in this data format and this reader only gives us the sentence.

I will fix this soon!

**Comment by hunterhector** [comment]
What's the progress on this?

**Comment by ziqian98** [comment]
> What's the progress on this?

Sorry for the late update. I am still testing and will update on this weekend.

**Comment by hunterhector** [comment]
updates?

**Comment by ziqian98** [comment]
> updates?

Something unexpected happened. Still fixing on it all these days.

**Comment by ziqian98** [comment]
I extracted the sentenceTokens filed, which is the original sentence, treat it as sentence. 
May I know if the following steps are right:

1. create [ "entry_name": "ft.onto.base_ontology.Question" ...] and [ "entry_name": "ft.onto.base_ontology.Answer" ...]under base_ontology.json
2. create a Question class and a Answer class under base_ontology.py

Using the Question class to read the question generated by the original sentence, and using the  Answer class to read the Answer generated by the original sentence.

**Comment by hunterhector** [comment]
Do you still want to finish this?

**Comment by ziqian98** [comment]
> Do you still want to finish this?

Sorry, I thought this reader was already completed by reading the sentence.
May I know what else should I read except the original sentence?

**Comment by ziqian98** [comment]
> Do you still want to finish this?

I talked with my teammates before but they are currently working on NER task and are not sure what to read either.

**Comment by hunterhector** [comment]
> > Do you still want to finish this?
> 
> I talked with my teammates before but they are currently working on NER task and are not sure what to read either.

The current implementation doesn't record any information from Question and Answer, can you make sure they are in?

**Comment by ziqian98** [comment]
Currently, I only read the original Sentence. I was puzzled about how to read the Questions and Answers in the previous comments. I will add this after I finish the Classification task, by the end of the weekday, and before they start the SRL task.

**Comment by hunterhector** [comment]
> Currently, I only read the original Sentence. I was puzzled about how to read the Questions and Answers in the previous comments. I will add this after I finish the Classification task, by the end of the weekday, and before they start the SRL task.

Yeah. Writing a reader requires you to understand the dataset (and a bit Forte). So you need to understand what this dataset is providing.

**Pull request closed without merging**